src/parser/jobs/blm/modules/Leylines.js

```diff
@@ -58,130 +53,104 @@ export default class Leylines extends Module {
 			order: 0,
 			nestedGroups: [],
 		})
+
 		this.timeline.addGroup(this._group)
 	}
 
-	// _onApplyBuff and _onLoseBuff retooled from RaidBuffs.js
-	_onApplyBuff(event) {
-		const buffs = this.getTargetBuffs(event)
-		const statusId = event.ability.guid
+	// Manage buff windows
+	_onGain(event) {
+		const status = getStatus(event.ability.guid)
 
-		// Make sure there's a nested group for us
-		const groupId = 'leybuffs-' + statusId
+		// Something is not right
+		if (!status) {
+			return
+		}
+
+		// Track the new window
+		const tracker = this._buffWindows[status.id]
+		tracker.current = {
+			start: event.timestamp,
+		}
+
+		// Manage timeline group
+		const groupId = 'leybuffs-' + status.id
 		if (!this._group.nestedGroups.includes(groupId)) {
 			this.timeline.addGroup(new Group({
 				id: groupId,
-				content: event.ability.name,
+				content: status.name,
 			}))
 			this._group.nestedGroups.push(groupId)
 		}
-
-		// Generate an item for the buff
-		const startTime = this.parser.fight.start_time
-		const status = STATUSES[statusId]
-		buffs[statusId] = new Item({
-			type: 'background',
-			start: event.timestamp - startTime,
-			group: groupId,
-			content: <img src={status.icon} alt={status.name}/>,
-		})
 	}
 
-	_onLoseBuff(event) {
-		const item = this.getTargetBuffs(event)[event.ability.guid]
-		// This shouldn't happen, but it do.
-		if (!item) { return }
+	_onDrop(event) {
+		this._stopAndSave(event.ability.guid, event.timestamp)
+	}
 
-		item.end = event.timestamp - this.parser.fight.start_time
-		this.timeline.addItem(item)
+	// We died, close windows
+	_onDeath(event) {
+		this._stopAndSave(STATUSES.LEY_LINES.id, event.timestamp)
 	}
 
-	_onApplyLeyLines(event) {
-		if (this._leyLines.current) { this._leyLines.history.push(this._leyLines.current) }
+	// Finalise a buff window
+	_stopAndSave(statusId, endTime = this.parser.currentTimestamp) {
+		const tracker = this._buffWindows[statusId]
 
-		this._leyLines.current = {
-			start: event.timestamp,
-			stop: null,
+		// Already closed, nothing to do here
+		if (!tracker.current) {
+			return
 		}
 
-		this._onApplyBuff(event)
-	}
+		// Close the window
+		tracker.current.stop = endTime
+		tracker.history.push(tracker.current)
+		tracker.current = null
 
-	_onRemoveLeyLines(event) {
-		if (this._leyLines.current) {
-			this._leyLines.current.stop = event.timestamp
+		// Close dependency windows
+		if (statusId === STATUSES.LEY_LINES.id) {
+			this._stopAndSave(STATUSES.CIRCLE_OF_POWER.id, endTime)
 		}
-		this._onLoseBuff(event)
 	}
 
-	_onApplyCircleOfPower(event) {
-		if (this._circleOfPowers.current) { this._circleOfPowers.history.push(this._circleOfPowers.current) }
-
-		this._circleOfPowers.current = {
-			start: event.timestamp,
-			stop: null,
-		}
-		this._onApplyBuff(event) // Also track application time for UI
+	// A reminder of man's ability to generate electricity
+	_dontMovePercent(power, lines) {
```

Comment:
\>`dontMovePercent`

Reply:
I was subjected to The Castle so many times during various years of schooling that it's permanently blacklisted from my brain.